consensus/consensus-types/src/wrapped_ledger_info.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::vote_data::VoteData;
+use anyhow::{ensure, Context};
+use aptos_types::{
+    aggregate_signature::AggregateSignature,
+    block_info::BlockInfo,
+    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+    validator_verifier::ValidatorVerifier,
+};
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+/// This struct is similar to QuorumCert, except that the verify function doesn't verify vote_data.
+/// vote_data and consensus_data_hash inside signed_ledger_info could be replaced with dummy values.
+#[derive(Deserialize, Serialize, Clone, Debug, Eq, PartialEq)]
+pub struct WrappedLedgerInfo {
+    /// The vote information certified by the quorum.
+    vote_data: VoteData,
+    /// The signed LedgerInfo of a committed block that carries the data about the certified block.
+    signed_ledger_info: LedgerInfoWithSignatures,
+}
+
+impl Display for WrappedLedgerInfo {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "WrappedLedgerInfo: [{}, {}]",
+            self.vote_data, self.signed_ledger_info
+        )
+    }
+}
+
+impl WrappedLedgerInfo {
+    pub fn new(vote_data: VoteData, signed_ledger_info: LedgerInfoWithSignatures) -> Self {
+        WrappedLedgerInfo {
+            vote_data,
+            signed_ledger_info,
+        }
+    }
+
+    #[cfg(any(test, feature = "fuzzing"))]
+    pub fn dummy() -> Self {
+        Self {
+            vote_data: VoteData::dummy(),
+            signed_ledger_info: LedgerInfoWithSignatures::new(
+                LedgerInfo::dummy(),
+                AggregateSignature::empty(),
+            ),
+        }
+    }
+
+    pub fn certified_block(&self) -> &BlockInfo {
```

Comment:
please remove any access to the vote data from this struct, it defeats the purpose of creating this struct for encapsulation

Reply:
I added order_vote_enabled flag in consensus_config. If this flag is set to false, the execution should happen similar to before. When order_vote_enabled flag is set to false, I was using certified_block in fast_forward_sync function in sync_manager.rs

Reply:
why this logic depends on the order_vote flag? isn't the idea to replace the quorum cert with wrapped ledger info and eliminate the use of vote data first?

Reply:
Removed it.